backend/cmm_helpers.ml

```diff
@@ -3345,7 +3345,120 @@ let trap_return arg trap_actions =
 
 let create_ccatch ~rec_flag ~handlers ~body =
   let rec_flag = if rec_flag then Cmm.Recursive else Cmm.Nonrecursive in
-  Cmm.Ccatch (rec_flag, handlers, body, Any)
+  let[@inline] default () = Cmm.Ccatch (rec_flag, handlers, body, Any) in
+  match body, handlers with
+  | ( Cifthenelse
+        ( Cop (Ccmpi Cge, [Cvar x; Cconst_int (1, _)], _),
```

Comment:
Since the comparison here is with `1` (i.e. tagged `0`), the optimization is only correct if we are sure that `x` can only hold a tagged integer, else an innocent comparison of unboxed integers between `1` and `n` (with `n` a positive compile-time constant), might be optimised and the semantics changed.

How do you ensure that `x` is necessarily a tagged integer here ?

Reply:
One solution proposed by @lthls is to instead do the folliwng translation:
`1 <= x {op} n` --> `x - 1 {unsigned_op} n - 1`

Reply:
(I have not spent enough time on it, but
it looks like there is a condition on `n`;
`min_int` would be problematic for instance.
But maybe I am confused between tagged
and untagged values.)

Reply:
@xclerc I'm not sure I follow: in this case, `n` is a compile-time constant literal, that must be positive for the optimisation to trigger, so it can never be `min_int` (and `n - 1` is also a literal that can be computed at compile-time)

Reply:
OK, I missed the condition about
`n` being positive.